server/grouping/grouping.go

```diff
@@ -138,12 +138,17 @@ func isInGroup(matchStatements []map[string]string, factmap map[string][]string)
 				continue
 			}
 
-			regexStrs := factmap["Hostname"]
-			for j := range matchStatements {
-				result := exp.Find([]byte(regexStrs[j]))
-				if result != nil {
-					return true
-				}
+			// retrieved hostname from system
+			//
+			// TODO(mierdin): Currently, hostname is provided as single
+			// string in slice, so we can go straight to zero index.
+			// This is due to crappy implementation in the fact collecting
+			// process, so this is an acceptable temporary workaround.
+			hostname := factmap["Hostname"][0]
+
+			result := exp.Find([]byte(hostname))
+			if result != nil {
+				return true
 			}
```

Comment:
This can be written as:
``` go
if exp.Match([]byte(hostname)) {
	return true
}
```

Reply:
Updated in https://github.com/toddproject/todd/pull/147/commits/16bf732b6f860db68dc3b1a55c981e945df2bad7